src/vaadin-context-menu.html

```diff
@@ -639,15 +640,28 @@
               // Preserve right-aligned, if possible.
               if ((wdthVport - (right - padding)) > width) {
                 overlay.setAttribute('right-aligned', '');
-                style.right = right - padding + 'px';
+                style.right = right + 'px';
                 alignedToParent = true;
               }
+            } else if (x < parentContentRect.x) {
+              // Check if sub menu opens on the left side and the parent menu is not right aligned.
+              // If so, use actual width of the submenu content instead of the parent menu content.
+              x = x - (width - parentContentRect.width);
             }
           }
 
           if (!alignedToParent) {
-            if (x < wdthVport / 2 || x < xMax) {
+            if ((x < wdthVport / 2 || x < xMax) && !parent) {
               style.left = x + 'px';
+            } else if ((x < wdthVport / 2 || x < xMax) && parent &&
+              wdthVport - parent.$.overlay.getBoundingClientRect().width - parent.$.overlay.getBoundingClientRect().x
```

Comment:
Please replace `parent.$.overlay.getBoundingClientRect()` with `parentContentRect` here and below.

Reply:
done

Reply:
I do not see any changes. Did you push the commit?

Reply:
I pushed it.